mne/preprocessing/detect_bad_channels.py

```diff
@@ -0,0 +1,67 @@
+# Authors: Velu Prabhakar Kumaravel <vkumaravel@fbk.eu>      
+# License: BSD-3-Clause
+
+
+import numpy as np
+
+from ..io.base import BaseRaw
+from .. import Transform
+from ..utils import (logger, verbose, _validate_type)
+
+
+@verbose
+def mark_bad_channels_lof(raw, n_neighbors = 8, metric = 'sqeuclidean', lof_threshold = 1.5):
```

Comment:
@vpKumaravel how much do you need to adjust the lof_threshold for different datasets? Can you report the list of bad channels you get on some public data so we can compare with what other methods report? If you take MEG data you can look at the difference between the list of bads you get here and what [find_bad_channels_maxwell](https://mne.tools/dev/generated/mne.preprocessing.find_bad_channels_maxwell.html#mne.preprocessing.find_bad_channels_maxwell) gives?

Reply:
Thanks a lot, @drammock and @agramfort for your comments.

1. I will ensure that format changes are fixed.
2. We validated the LOF algorithm in Newborn, Infant, and Adult EEG and compared the results with different measures used in EEGLAB. The results are published [here](https://www.mdpi.com/1424-8220/22/19/7314/htm).
3. As far as the LOF threshold is concerned, our results suggest that if the dataset has a relatively higher SNR, a strict threshold of around 1.5 is optimal. For very noisy datasets (like the ones from Newborns), a relaxed threshold of around 2.5 is optimal. Here is the [figure ](https://www.mdpi.com/sensors/sensors-22-07314/article_deploy/html/images/sensors-22-07314-g006.png) from the paper.

If you would like to see how bad channel methods in MNE perform compared to the LOF, I will be happy to do that. Also testing LOF in MEG could be interesting. Do you perhaps know of any open source datasets with annotated bad channels? I did a search on OpenNeuro but I cannot find any so far! @agramfort